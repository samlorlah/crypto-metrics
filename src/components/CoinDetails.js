import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchSingleCoin } from '../redux/coinList/coinList';
import styles from './CoinDetails.module.css';

function CoinDetails() {
  const { detailsList } = styles;
  const { id } = useParams();
  const coin = useSelector((state) => state.coins.coinDetails);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchSingleCoin(id));
  }, []);
  return (
    <>
      <ul className={detailsList}>
        <li>
          <h5>Name</h5>
          <p>{coin.name}</p>
        </li>
        <li>
          <h5>Symbol</h5>
          <p>{coin.symbol}</p>
        </li>
        <li>
          <h5>Price</h5>
          <p>
            ~ $
            { parseFloat(coin.priceUsd).toLocaleString('en-US', { minimumFractionDigits: 2, maximumFractionDigits: 2 }) }
          </p>
        </li>
        <li>
          <h5>% Change (24 hrs)</h5>
          <p>
            {coin.changePercent24Hr}
          </p>
        </li>
        <li>
          <h5>Supply</h5>
          <p>
            {coin.supply}
          </p>
        </li>
        <li>
          <h5>VWAP (24 hrs)</h5>
          <p>
            ~ $
            { parseFloat(coin.vwap24Hr).toLocaleString('en-US', { minimumFractionDigits: 2, maximumFractionDigits: 2 }) }
          </p>
        </li>
      </ul>
    </>
  );
}

export default CoinDetails;
