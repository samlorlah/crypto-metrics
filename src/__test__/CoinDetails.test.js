import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import CoinDetails from '../components/CoinDetails';

describe('Test Compenent that renders all the lists of coins', () => {
  it('get the list of coins from store and display', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <CoinDetails />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
