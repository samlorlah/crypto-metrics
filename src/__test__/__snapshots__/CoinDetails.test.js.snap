// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Compenent that renders all the lists of coins get the list of coins from store and display 1`] = `
<ul
  className="detailsList"
>
  <li>
    <h5>
      Name
    </h5>
    <p />
  </li>
  <li>
    <h5>
      Symbol
    </h5>
    <p />
  </li>
  <li>
    <h5>
      Price
    </h5>
    <p>
      ~ $
      NaN
    </p>
  </li>
  <li>
    <h5>
      % Change (24 hrs)
    </h5>
    <p />
  </li>
  <li>
    <h5>
      Supply
    </h5>
    <p />
  </li>
  <li>
    <h5>
      VWAP (24 hrs)
    </h5>
    <p>
      ~ $
      NaN
    </p>
  </li>
</ul>
`;
